lib/active_merchant/billing/gateways/midtrans.rb

```diff
@@ -267,10 +268,27 @@ def add_authorization(post, money, authorization)
 
       def contruct_refund_request(post, money, authorization, options={})
         post[:transaction_id] = authorization
+        post[:rail_code] = options[:rail_code] if options[:rail_code]
         post[:details] = {}
         post[:details][:amount] = money if money
         post[:details][:reason] = options[:reason] if options[:reason]
-        post[:details][:refund_key] = options[:order_id] if options[:order_id]
+        post[:details][:refund_key] = options[:refund_transaction_id] if options[:refund_transaction_id]
+      end
+
+      def handle_direct_refund(transaction_id, payload)
+        @uri = URI.parse("#{url()}/v2/#{transaction_id}/refund/online/direct")
+        begin
+          https = Net::HTTP.new(url.host, url.port)
+          request = Net::HTTP::Post.new(url)
+          request["Accept"] = "application/json"
+          request["Content-Type"] = "application/json"
+          request["Authorization"] = "Basic #{Base64.strict_encode64(@options["server_key"] + ':')}"
+          request.body = JSON.dump(payload)
+          response = https.request(request)
+          JSON.parse(response.body)        
+        rescue ResponseError => e
+          Response.new(false, e.response.message)

```

Comment:
is this method not available in the gateway?

Reply:
No this method is not yet available in the `veritrans` gem